### PR TITLE
Condenses wigs into one object

### PR DESCRIPTION
**Pull request by KoboldCommando**
# About The Pull Request

There used to be wigs, random wigs, and natural wigs.  I've seen it confuse people and have been confused by it several times.

Now, there is only one wig to rule them all!  It spawns with a random style and color.  When you use it in hand, it uses a radial menu to allow you to restyle, recolor, and switch to and from natural mode, which will mimic the wearer's natural hair color.

I also did a few tweaks.  The original sprite icons had a weird layering problem and used sprite overlays to create their inventory sprites.  Now they just use, well, the icons!  There was also a check to prevent bald wigs having invisible icons that constructed a list then subtracted from it.  Now it just checks the style directly later, and checks for both bald styles.

All previous locations of wigs, barber spawns, the clothes vendor, the biogenerator, should now be stocked with these unified wigs.

## Why It's Good For The Game

(hopefully) less obtuse mechanics, fewer redundant items, and MORE FASHION!

## Changelog

:cl:
refactor: There's now only one type of wig, which can change style, color, and copying the wearer's hair color, using a radial menu!
/:cl:

### CODE REVIEW

**Comment by KoboldCommando** [comment]
I'm... not sure why that rogue merge commit snuck in there lol.  Hopefully it doesn't cause any problems.

File: code/modules/clothing/head/misc_special.dm [hunk 1]
```diff
@@ -238,22 +238,31 @@
...
 	else
-		var/mutable_appearance/M = mutable_appearance(S.icon,S.icon_state)
-		M.appearance_flags |= RESET_COLOR
-		M.color = hair_color
-		add_overlay(M)
+		icon = S.icon
+		icon_state = S.icon_state
+	color = hair_color
+
+/obj/item/clothing/head/wig/equipped(mob/living/carbon/human/user, slot)
```
Comment (nednaZ) [inline code/modules/clothing/head/misc_special.dm:260]: Natural hair mode appears to default to pure white and never changes.

**Review by nednaZ — CHANGES_REQUESTED** [review]
All other aspects appear to work.
I believe the issue with the natural hair mode is that it requires a 6 digit hexcode but gets a 3 digit one from the user's hair_color variable.

File: code/modules/clothing/head/misc_special.dm (hunk 1 above)
Comment (KoboldCommando) [reply code/modules/clothing/head/misc_special.dm:260]: It was actually simpler than that!  Funnily enough I attempted to solve it with regex before realizing how straightforward it actually was.

Byond seems to auto-convert between 3-digit and 6-digit color codes.  The problem is that I was setting the same variable that update_icon updates.  It should work now!

**Review by nednaZ — APPROVED** [review]
Looks good to me.

A funny side-effect with Natural Mode is that throwing it onto another person works too.
Some people might not even notice it happen because of that.

**Pull request merged**